tests/managedmcg_controller_test.go

```diff
@@ -6,45 +6,189 @@ import (
+		deadMansSecretfake.Data["SNITCH_URL"] = []byte("foo")
+
+		smtpSecretFake := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "smtp-secret-fake",
+				Namespace: namespace,
+			},
+		}
+		smtpSecretFake.Data = make(map[string][]byte, 1)
+		smtpSecretFake.Data["host"] = []byte("host")
+		smtpSecretFake.Data["port"] = []byte("8080")
+		smtpSecretFake.Data["username"] = []byte("username")
+		smtpSecretFake.Data["password"] = []byte("password")
+
+		ocscsv := opv1a1.ClusterServiceVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ocs-operator",
+				Namespace: namespace,
+			},
+		}
+
+		noobacsv := noobaav1alpha1.NooBaa{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "nooba-operator",
+				Namespace: namespace,
+			},
+		}
+
+		BeforeEach(func() {
+			r.Scheme = k8sClient.Scheme()
+			r.Client = fake.NewClientBuilder().WithScheme(k8sClient.Scheme()).WithObjects(newNamespace, newManagedMCG, &ocscsv, &smtpSecretFake, &deadMansSecretfake, &pagerDutySecretFake, &noobacsv).Build()
+		})
+
+		AfterEach(func() {
+			err := r.Client.Get(context.Background(), client.ObjectKeyFromObject(&ocscsv), &ocscsv)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Delete(context.Background(), &ocscsv, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(&smtpSecretFake), &smtpSecretFake)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Delete(context.Background(), &smtpSecretFake, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(&deadMansSecretfake), &deadMansSecretfake)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Delete(context.Background(), &deadMansSecretfake, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(&pagerDutySecretFake), &pagerDutySecretFake)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Delete(context.Background(), &pagerDutySecretFake, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(&noobacsv), &noobacsv)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Delete(context.Background(), &noobacsv, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(newManagedMCG), newManagedMCG)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Delete(context.Background(), newManagedMCG, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(newNamespace), newNamespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = r.Client.Delete(context.Background(), newNamespace, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Same as `BeforeEach` comment above, but for deletion.

Reply:
yes ,I will remove the before check from here as well. Thanks